corehq/apps/locations/views.py

```diff
@@ -360,6 +360,20 @@ def archive_location(request, domain, loc_id):
         )
     })
 
+@can_edit_location
+def delete_location(request, domain, loc_id):
```

Comment:
You should be able to do this: https://github.com/dimagi/commcare-hq/blob/91f8a3e6fbefc223a411773020cabd0556c4a140/corehq/apps/reports/views.py#L723  (That's how the django decorators `@require_POST` and `@require_GET` are implemented)